sale_coupon_financial_risk/models/sale_order.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Tecnativa - Stefan Ungureanu
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def action_confirm(self):
+        exception_msg = self.evaluate_risk_message(
+            self.partner_invoice_id.commercial_partner_id
+        )
+        if exception_msg:
```

Comment:
If the risk is bypassed we should send the mail (test this as well)
```suggestion
        if exception_msg and not self.env.context.get("bypass_risk"):
```